card-component-app/UserCard.jsx

```diff
@@ -0,0 +1,28 @@
+const UserCard = (props) => {
+  const [email, setEmail] = React.useState('loading...')
+  const [firstname, setFirstName] = React.useState('loading...')
+  const [lastname, setLastName] = React.useState('loading...')
+  const [avatar, setAvatar] = React.useState('...loading')
+
+  React.useEffect(() => {
+    async function userInfo(userId) {
```

Comment:
👍 to making a named function for this fetching. I think this works well too. You could also define this outside the component entirely and then use a `then` callback to do the setting of state, but either way works!